### ISSUE TITLE
option xsize has no effect with certain commands and options (fastx_filter and chimeras)

### ISSUE THREAD

**Issue by frederic-mahe** [torognes/vsearch#272]
According to the `vsearch -h` command, the option `--xsize` removes the abundance annotation from fastq/fasta headers. But it has no effect when used with `--fastx_filter` (with vsearch 2.5.0):
```sh
vsearch --fastx_filter <(printf "@s;size=1;\nA\n+\nI\n") --xsize --fastqout - 2> /dev/null
```
returns:
```
@s;size=1;
A
+
I
```

**Comment by gregcaporaso** [comment]
I'm also noticing this with the `--chimera` output (but not the `--nonchimera` output) for both `--uchime_ref` and `--uchime_denovo`.

```bash
$ vsearch --uchime_ref    seqs-w-sizes.fasta --uchimeout out.tsv \
 --nonchimeras nonchim.fna --chimeras chime.fna --dn 1.4 --mindiffs 3 --mindiv 0.8 \
 --minh 0.28 --xn 8.0 --xsize --qmask none --db 76-otus/dna-sequences.fasta --quiet

$ head -1 chime.fna
>chime2;size=1

$ head -1 nonchim.fna
>eb3564a35320b53cef22a77288838c7446357327
```

```bash
$ vsearch --uchime_denovo    seqs-w-sizes.fasta --uchimeout out.tsv \
 --nonchimeras nonchim.fna --chimeras chime.fna --dn 1.4 --mindiffs 3 \
 --mindiv 0.8 --minh 0.28 --xn 8.0 --xsize --qmask none --quiet

$ head -1 chime.fna
>chime1;size=1

$ head -1 nonchim.fna
>e84fcf85a6a4065231dcf343bb862f1cb32abae6
```

**Comment by torognes** [comment]
I plan to fix this on Monday 18 Dec.

**Comment by torognes** [comment]
The reported problems should be fixed in release 2.6.2. I suspect the problems may apply to other commands than fastx_filter, uchime_denovo and uchime_ref, so I will keep the issue open until I have checked that out. The code involved also needs some refactoring.

**Comment by gregcaporaso** [comment]
Thanks for the quick response @torognes!

**Comment by torognes** [comment]
Fixed in version 2.7.0 just released.
